packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -566,6 +566,81 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     }).then<int>((_) => count);
   }
 
+  /// Triggers multiple frames for `duration` amount of time or until there are
+  /// no longer any frames scheduled, whichever is earlier.
+  ///
+  /// This does not promise to exhaust all pending frames. There maybe remaining
+  /// animation frames and therefore undisposed `Ticker`s.
+  ///
+  /// This will call [pump] at least once, even if no frames are scheduled when
+  /// the function is called, to flush any pending microtasks which may
+  /// themselves schedule a frame.
+  ///
+  /// The `frameRefreshRate` specify the frequency (in Hz) for frame requests.
+  /// This is an analog to screen refresh rate on a device. The frequency rate
+  /// will be rounded down to integer microseconds frame interval.
+  /// Default value is 59.94 Hz, which is the real refresh rate for many devices
+  /// when it shows as 60 Hz, for historical reason.
+  ///
+  /// This allows a test to pump for a period of time during an unbounded
+  /// animation. To pump to the end of a bounded animation, use [pumpAndSettle]
+  /// instead.
+  ///
+  /// If the function returns, it returns the number of pumps that it performed.
+  /// When building and rendering frames is not expensive, the return value
+  /// should be `1 + (duration.inMicroseconds / (1E6 ~/ frequency)).floor()`
+  /// or `1 + (\[remaining animation time\] / (1E6 ~/ frequency)).ceil()`.
+  /// When using the return number as a test result, avoid using values that
+  /// makes the devision above to be very close to an integer. This may causing
+  /// flaky result result due to time fluctuation and rounding error.
+  ///
+  /// Example:
+  ///     testWidgets('pumpContinuous', (WidgetTester tester) async {
```

Comment:
examples should use the snippet mechanism so that it gets syntax highlighting and `flutter create` templates and so on.

Reply:
(and most importantly so that it gets analyzed in CI)

Reply:
Can I have an example how this is done in `flutter_test`? The only example code in doc string I can find in `flutter_test` is https://api.flutter.dev/flutter/flutter_test/benchmarkWidgets.html . This example is almost a copy of my unit test.

Reply:
> Can I have an example how this is done in `flutter_test`? The only example code in doc string I can find in `flutter_test` is https://api.flutter.dev/flutter/flutter_test/benchmarkWidgets.html . This example is almost a copy of my unit test.

Problem solved.